File: spring-cloud-gateway/gateway-starter/src/main/java/com/livk/cors/CorsConfig.java
package com.livk.cors;

import com.livk.auto.service.annotation.SpringAutoService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.reactive.CorsWebFilter;
import org.springframework.web.cors.reactive.UrlBasedCorsConfigurationSource;

import java.util.Collections;

/**
 * <p>
 * CorsConfig
 * </p>
 *
 * @author livk
 * @date 2022/9/27
 */
@Configuration
@SpringAutoService
public class CorsConfig {

    @Bean
    public CorsWebFilter corsWebFilter() {
        CorsConfiguration corsConfiguration = new CorsConfiguration();
        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        corsConfiguration.addAllowedHeader("*");
        corsConfiguration.addAllowedMethod("*");
        corsConfiguration.setAllowedOriginPatterns(Collections.singletonList("*"));
        corsConfiguration.setAllowCredentials(true);
        source.registerCorsConfiguration("/**", corsConfiguration);
        return new CorsWebFilter(source);
    }
}

File: spring-cloud-gateway/gateway-starter/src/main/kotlin/com/livk/cors/CorsConfig.kt
package com.livk.cors

import com.livk.auto.service.annotation.SpringAutoService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.reactive.CorsWebFilter
import org.springframework.web.cors.reactive.UrlBasedCorsConfigurationSource

/**
 * @author livk
 */
@Configuration
@SpringAutoService
open class CorsConfig {
    @Bean
    open fun corsWebFilter(): CorsWebFilter {
        val corsConfiguration = CorsConfiguration()
        val source = UrlBasedCorsConfigurationSource()
        corsConfiguration.addAllowedHeader("*")
        corsConfiguration.addAllowedMethod("*")
        corsConfiguration.setAllowedOriginPatterns(listOf("*"))
        corsConfiguration.allowCredentials = true
        source.registerCorsConfiguration("/**", corsConfiguration)
        return CorsWebFilter(source)
    }
}
